beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -279,28 +301,76 @@ func (s *Service) deleteBlockFromPendingQueue(slot uint64, b *ethpb.SignedBeacon
 		newBlks = append(newBlks, blk)
 	}
 	if len(newBlks) == 0 {
-		delete(s.slotToPendingBlocks, slot)
-		return
+		s.slotToPendingBlocks.Delete(slotToCacheKey(slot))
+		return nil
+	}
+
+	// Decrease exp itme in proportion to how many blocks are still in the cache for slot key.
+	d := gcache.DefaultExpiration.Seconds() / float64(len(newBlks))
+	if err := s.slotToPendingBlocks.Replace(slotToCacheKey(slot), newBlks, time.Duration(d)); err != nil {
```

Comment:
so `d` is in terms of seconds while duration is in term of nanoseconds. The unit of account needs to be fixed here.

Reply:
are you sure? `gcache.DefaultExpiration.Seconds()` is already in ns representation

![Screen Shot 2020-11-18 at 7 52 21 AM](https://user-images.githubusercontent.com/21316537/99553439-fe09d580-2972-11eb-9516-15c6a15fad8f.png)

Reply:
https://golang.org/pkg/time/#Duration.Seconds

When calling the `Seconds` method it returns a float64